lib/archethic/metrics/poller.ex

```diff
@@ -23,45 +21,21 @@ defmodule ArchEthic.Metrics.Poller do
     Process.send_after(self(), {:periodic_calculation_of_points}, 10_000)
   end
 
-  def send_updates(%{data: data, pid_refs: pid_refs}) do
-    pid_refs
-    |> Task.async_stream(fn {pid_k, _pid_v} -> send(pid_k, {:update_data, data}) end)
-    |> Stream.run()
-  end
-
   def monitor() do
     GenServer.call(__MODULE__, :monitor)
   end
 
-  def handle_call(:monitor, {pid, __tag}, state) do
-    new_state = %{state | pid_refs: Map.put(state.pid_refs, pid, nil)}
-    send_updates(new_state)
-    _mref = Process.monitor(pid)
-    {:reply, :ok, new_state}
+  def handle_call(:monitor, {pid, _tag}, state) do
+    {:reply, :ok, Helpers.register_process(pid, state)}
   end
 
   def handle_info({:DOWN, _ref, :process, from_pid, _reason}, state) do
-    {_removed_pid, new_pid_refs} = Map.pop(state.pid_refs, from_pid)
-    new_state = %{state | pid_refs: new_pid_refs}
-    {:noreply, new_state}
+    {:noreply, Helpers.deregister_process(from_pid, state)}
   end
 
   def handle_info({:periodic_calculation_of_points}, current_state) do
-    new_state =
-      case Enum.empty?(current_state.pid_refs) do
-        false ->
-          send_updates(current_state)
-          %{data: get_new_data(), pid_refs: current_state.pid_refs}
-
-        true ->
-          current_state
-      end
-
+    new_state = Helpers.process_new_state(current_state)
```

Comment:
Hi ,
@samuel-uniris 
How can I test for external dependency which is in another module and being called by genserver Poller

Reply:
def process_new_state(current_state) do
    case Enum.empty?(current_state.pid_refs) do
      false ->
        dipatch_updates(current_state)
        %{data: network_collector(), pid_refs: current_state.pid_refs}

      true ->
        current_state
    end
  end